### PR TITLE
Sort Dropdown Fixes

### LINKED ISSUE #597: Podcast sort dropdown fixups
Couple minor things to fix on the Podcast#index sort picker.

image

- [x] Don't underline on hover/focus
- [x] Decide/design some UI for how the _current_ sort order is indicated.  Currently you'd have to expand the dropdown to see your current sort order.  But probably some better/other options.
- [x] Implement the ruby code to indicate what the current sort is (adding the `active` class, or whatever)

### PR DESCRIPTION
**Pull request by a-maci29**
Closes #597

### CODE REVIEW

File: app/controllers/podcasts_controller.rb [hunk 1]
```diff
@@ -4,14 +4,16 @@ class PodcastsController < ApplicationController
...
+                   "Recent Activity" => {updated_at: :desc}}.freeze
 
   DEFAULT_PAGE_SIZE = 10
 
   # GET /podcasts
   def index
     base_query = policy_scope(Podcast).page(params[:page]).per(DEFAULT_PAGE_SIZE).includes(default_feed: :feed_images)
-    @podcasts = add_sorting(base_query).filter_by_title(params[:q])
+    filtered_podcasts = base_query.filter_by_title(params[:q])
+    @podcasts = add_sorting(filtered_podcasts)
```
Comment (cavis) [inline app/controllers/podcasts_controller.rb:16]: Noting this change is functionally equivalent to what was there before.  (Calling both `add_sorting` and `.filter_by_title` on the base query, assigning the result to `@podcasts`).  But this is fine, if you think it's more readable this way.

File: app/views/podcasts/index.html.erb [hunk 2]
```diff
@@ -6,15 +6,23 @@
...
+          <% else %>
+            <span class="material-icons" aria-hidden="true">timer</span>
+          <% end %>
         </button>
         <ul class="dropdown-menu">
-          <li><%= link_to "Recent Activity", podcasts_path(sort: "Recent"), class: "dropdown-item active" %></li>
-          <li><%= link_to "# of Episodes", podcasts_path(sort: "episode_count"), class: "dropdown-item" %></li>
-          <li><%= link_to "A-Z", podcasts_path(sort: "A-Z"), class: "dropdown-item" %></li>
-          <li><%= link_to "Z-A", podcasts_path(sort: "Z-A"), class: "dropdown-item" %></li>
+        <li><%= active_link_to "Recent Activity", podcasts_path(sort: "Recent Activity"), active: :exact, class: "dropdown-item text-decoration-none" %></li>
```
Comment (cavis) [inline app/views/podcasts/index.html.erb:22]: I think these should stay indented, inside the `<ul>`?

**Review by cavis — APPROVED** [review]
One nitpick on indentation.  But otherwise looks good to me.

Should probably close #597 and mark this as a non-draft?  I got them mixed up since this one is still a draft.

File: app/controllers/podcasts_controller.rb (hunk 1 above)
Comment (a-maci29) [reply app/controllers/podcasts_controller.rb:16]: The latter way would maybe be easier for me to come back and understand what's happening, but I'm not insistent about it. I can change it back to the way it was before.

File: app/views/podcasts/index.html.erb (hunk 2 above)
Comment (a-maci29) [reply app/views/podcasts/index.html.erb:22]: Huh. It's indented in my VS code. Would linter change that??

**Review by cavis — APPROVED** [review]
👍 looks good to me!

**Pull request merged**